docs/DeveloperGuide.md

```diff
@@ -154,6 +154,38 @@ Classes used by multiple components are in the `ay2122s1_cs2103t_w16_2.btbb.comm
 
 This section describes some noteworthy details on how certain features are implemented.
 
+
+### List feature
+
+#### Overview
+
+List operation can be executed for all entity types in the model,
+ie. clients, orders, ingredients and recipes.
+
+#### Mechanism
+This section explains the List Order Mechanism:
+1. The user executes the `ListOrderCommand` to display everything stored in the specified list.
+2. The `ListOrderCommandParser` parses the command into a `ListOrderCommand`, which is passed back to the logic to be executed.
+3. The `Logic` Component then executes the command.
+4. The Model's `filteredOrderList` will be updated to show all orders.
+5. The `CommandResult` created would then be passed to the Ui components, to display the updated order list and result message to the user.
+
+This Mechanism is the same for all other entities (client, ingredient and recipe) in the model.
+
+The following sequence diagram shows how the list operation works:
+![ListSequenceDiagram](images/ListSequenceDiagram.png)
```

Comment:
Over here, I think you should try to give a general explanation on how the mechanism works for all entity types not just order. You can move the current explanation to the usage scenario section  below as it will be easier to explain how list order works in detail in a usage scenario. You can look at my PR for reference.